### PR TITLE
Collections Lab

### PR DESCRIPTION
**Pull request by ewashbourne**
{
  "comfortability": 3,
  "completeness": 5 (I believe it's complete),
  "questions": No questions at this time.
}

### CODE REVIEW

**Comment by rgpass** [comment]
Remember to assign one of us.

File: talkative_robot_2.rb [hunk 1]
```diff
@@ -29,9 +29,9 @@ def bears_giraffes
 
 name 				= get_user_name
 age 				= get_user_age
-gender 			= get_user_gender
-hours_slept = hours_of_sleep      
-animals			= bears_giraffes 
+gender 				= get_user_gender
+hours_slept 		= hours_of_sleep      
+animals				= bears_giraffes
```
Comment (rgpass) [inline talkative_robot_2.rb]: Fancy

File: talkative_robot_2.rb [hunk 2]
```diff
@@ -87,3 +87,71 @@ def bears_or_giraffes(b_or_g)
 end
 
 bears_or_giraffes(animals)
+
+
+# ---------- Below is the lab for arrays and hashes ---------- #
```
Comment (rgpass) [inline talkative_robot_2.rb]: Remove comments if they're not adding any value.

File: talkative_robot_2.rb [hunk 3]
```diff
@@ -87,3 +87,71 @@ def bears_or_giraffes(b_or_g)
...
+# Add new item to list “Oh yeah, don’t forget the bread!”
+
+grocery_list = ["eggs", "bread", "orange juice", "potatoes", "peanut butter"]
+
+user = { }
+user[:Name] 							= name
+user[:Age] 								= age
+user[:Gender] 						= gender
+user[:Hours_Slept] 				= hours_slept
+user[:Favorite_Animal] 		= animals
```
Comment (rgpass) [inline talkative_robot_2.rb]: Hard to tell on GitHub, but it looks like the `=` signs are aligned. If so, yay! If not, would be better to do so.

File: talkative_robot_2.rb [hunk 4]
```diff
@@ -87,3 +87,71 @@ def bears_or_giraffes(b_or_g)
...
+# from the array -- otherwise the it remains. In the end, the revised list is printed.
+
+
+grocery_list = ["eggs", "bread", "chicken", "potatoes", "milk"]
+
+puts "Your grocery list:"
+puts grocery_list
+
+index = 4
+while index >= 0
```
Comment (rgpass) [inline talkative_robot_2.rb]: Going ahead, nice! It's better to have it be `while something do` -- remember the `do`

File: talkative_robot_2.rb [hunk 5]
```diff
@@ -87,3 +87,71 @@ def bears_or_giraffes(b_or_g)
...
+	print "Did you remember to grab the #{grocery_list[index]}. Y/N? "
+	user_input = gets.chomp.upcase
+		if user_input == "Y"
+			grocery_list.delete_at(index)	
+		end
+	index = index - 1
```
Comment (rgpass) [inline talkative_robot_2.rb]: Can clean this up with either `-=` or `--`

File: talkative_robot_2.rb [hunk 6]
```diff
@@ -87,3 +87,71 @@ def bears_or_giraffes(b_or_g)
...
+end
+
+puts "Here is your revised grocery list: " 
+puts grocery_list
+
+# This is a loop where the user is asked if he/she would like to add anything 
+# to the list. If yes, the user can add an item, which is pushed to grocery_list.
+# If the user selects no, the loop ends.
+
+while true
```
Comment (rgpass) [inline talkative_robot_2.rb]: While loops need an incrementer or some sort of sign of when to stop. I see you have a break in there, but usually it's better to have a variable (even if it's set to true)... that way it reads nicer.

**Comment by rgpass** [comment]
:+1:

**Comment by ewashbourne** [comment]
As always, thanks for the feedback.

When I finished the code in Sublime, I had it nicely formatted (i.e. things
lined up/visually organized), but when I pushed it to GitHub, certain parts
were changed. I think it has something to do with setting the tab correctly
(two spaces instead of defaulting to four).

**Comment by ewashbourne** [comment]
As always, thanks for the feedback. 

When I finished the code in Sublime, I had it nicely formatted (i.e. things lined up/visually organized), but when I pushed it to GitHub, certain parts were changed. I think it has something to do with setting the tab correctly  in Sublime (two spaces instead of defaulting to four).

**Comment by ewashbourne** [comment]
I think I fixed the problem with the default tab setting in SublimeText...

**Comment by ewashbourne** [comment]
@rgpass

**Pull request closed without merging**